File: backend/rm_service/schemas/medical_record.py
"""
Medical Record Schemas
"""
from pydantic import BaseModel
from datetime import datetime
from typing import Optional, List
from enum import Enum

class VisitTypeEnum(str, Enum):
    OUTPATIENT = "outpatient"
    INPATIENT = "inpatient"
    EMERGENCY = "emergency"

class MedicalRecordCreate(BaseModel):
    """Schema for creating medical record - for backoffice (petugas), patient_id must be provided"""
    patient_id: int  # Required for backoffice - petugas can create records for any patient
    visit_date: datetime
    visit_type: VisitTypeEnum
    diagnosis_summary: Optional[str] = None
    notes: Optional[str] = None
    doctor_name: Optional[str] = None
    facility_name: Optional[str] = None

class MedicalRecordResponse(BaseModel):
    """Schema for medical record response"""
    record_id: str
    patient_id: int  # FK to users.id
    visit_date: datetime
    visit_type: str
    diagnosis_summary: Optional[str] = None
    notes: Optional[str] = None
    doctor_name: Optional[str] = None
    facility_name: Optional[str] = None
    created_at: datetime
    
    class Config:
        from_attributes = True

class MedicalRecordFull(MedicalRecordResponse):
    """Full medical record with related data"""
    diagnoses: Optional[List] = []
    prescriptions: Optional[List] = []
    lab_results: Optional[List] = []

File: apps/lib/domain/entities/medical_record.dart
/// Medical Record entity - Business object untuk rekam medis
class MedicalRecord {
  final String recordId;
  final int patientId;
  final DateTime visitDate;
  final String visitType; // outpatient, inpatient, emergency
  final String? diagnosisSummary;
  final String? notes;
  final String? doctorName;
  final String? facilityName;
  final DateTime createdAt;
  
  const MedicalRecord({
    required this.recordId,
    required this.patientId,
    required this.visitDate,
    required this.visitType,
    this.diagnosisSummary,
    this.notes,
    this.doctorName,
    this.facilityName,
    required this.createdAt,
  });
}

